src/Components/ShortcutKeyModal/SearchModal.js

```diff
@@ -0,0 +1,187 @@
+  const [selectItem, setSelectItem] = useState(0);
+  const [users, setUsers] = useState([]);
+  const { user } = useUser();
+
+  const routes = useMemo(() => {
+    if (user.accessLevel === membershipState.MEMBER) return [...memberSignedInRoutes, ...signedOutRoutes];
+    if (user.accessLevel >= membershipState.OFFICER) return [...officerSignedInRoutes, ...memberSignedInRoutes, ...signedOutRoutes, ...users];
+    return [...signedOutRoutes];
+  }, [user.accessLevel, users]);
+
+  function handleChanges(e) {
+    setKeyword(e.target.value);
+    setSelectItem(0);
+  }
+
+  async function getUserData() {
+    try {
+      const apiResponse = await getAllUsers({
+        token: user.token,
+        query: keyword,
+        page: 0,
+        sortColumn: 'firstName',
+        sortOrder: 'asc'
+      });
+      setUsers(apiResponse.responseData.items);
+      // console.log('api fetch') // For debug
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  /**
+   * An effect that instantly shows all hardcoded routes.
+   * @dependencies keyword, routes, open
+   */
+  useEffect(() => {
+    if (!open) return;
+
+    // Instantly display for the hardcoded page recommendations
+    if (keyword) {
+      const routeMatches = routes.filter((r) =>
+        r.pageName?.toLowerCase().includes(keyword.toLowerCase())
+      );
+      setSuggestions(routeMatches);
+    } else setSuggestions([]);
+  }, [open, keyword, routes]);
+
+  /**
+   * A debounce function that performs the search 400ms after the user stops typing.
+   * @dependencies keyword, open, user.accessLevel
+   */
+  useEffect(() => {
+    if (!open ||
+      !user.accessLevel ||
+      user?.accessLevel < membershipState.OFFICER ||
+      !keyword) return;
+
+    const debounce = setTimeout(() => {
+      // Only fetch users when there is a change in keyword
+      if (prevKeyword.current !== keyword) {
+        getUserData();
+        prevKeyword.current = keyword; // Update previous keyword after fetching for new data
+      }
+    }, 400);
+
+    return () => clearTimeout(debounce);
+  }, [keyword, open, user.accessLevel]);
+
+  /**
+   * Combines hardcoded route suggestions with user search results
+   * after the debounced fetch has updated the user list.
+   * Only runs when the user list is updated, and search is open.
+   * @dependencies open, users, keyword, routes, user.accessLevel
+   */
+  useEffect(() => {
+    if (!open ||
+      !user.accessLevel ||
+      user.accessLevel < membershipState.OFFICER ||
+      !keyword) return;
+
+    const userMatches = users.filter((u) => {
+      const searchKey = keyword.toLowerCase();
+      return (
+        u.firstName?.toLowerCase().includes(searchKey) ||
+        u.lastName?.toLowerCase().includes(searchKey) ||
+        u.email?.toLowerCase().includes(searchKey)
+      );
+    }).map((u) => ({
+      pageName: `${u.firstName} ${u.lastName} (${u.email})`,
+      path: `/user/edit/${u._id}`,
```

Comment:
what if we include the full url, maybe it looks ok?
```suggestion
      path: `${window.location.origin}/user/edit/${u._id}`,
```